File: prime_fibonacci.cpp
#include <bits/stdc++.h>
#include <cstdlib>
#include <math.h>
#include <cmath>
#include <string>

#define MAX 1000005
#define ll long long
using namespace std;

bool isPrime(ll num)
{
    if(num < 2)
        return(false);

    for(ll i=2; i<sqrt(num)+1; i++)
    {
        if(num%i == 0)
            return(false);
    }
    return(true);

}
int main()
{
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

    int primes1[1000], primes2[1000];
    int n1,n2;
    int ans;
    cin>>n1>>n2;
    int pos = 0;
    int k = 0;
    for(int i=n1; i<=n2; i++)
    {
        if(isPrime(i))
            primes1[pos++] = i;
    }

    if(pos == 0)
        ans = 0;

    else
    {
        for(int i=0; i<pos; i++)
        {
            for(int j=0; j<pos; j++)
            {
                if(i!=j)
                {
                    string a = to_string(primes1[i]);
                    string b = to_string(primes1[j]);
                    a = a+b;
                    int number = stoi((a));
                    cout<<number<<endl;
                    if(isPrime(number))
                        primes2[k++] = number;
                }
            }
        }

        std::set<int>s(begin(primes2), end(primes2));

        if(s.size() == 0)
            ans = 0;
        else
        {
            ll x = *s.begin();
            ll y = *s.rbegin();
            ll z = 0;
            for(ll i=2; i<s.size(); i++)
            {
                z = x+y;
                x = y;
                y = z;
            }
            ans = z;
        }
    }

    cout<<ans<<endl;

    return 0;

}

File: prime_fibonacci.py
# import math
#
# def isPrime(num):
#     if(num < 2):
#         return(False)
#
#     for i in range(2, int(math.sqrt(num))+1):
#         if(num % i == 0):
#             return(False)
#     return(True)
#
# def fibonacci(a,b,length):
#     c = 0
#     for i in range(2,length):
#         c = a + b
#         a = b
#         b = c
#     return(c)
#
# line = input().split()
# n1,n2 = int(line[0]),int(line[1])
#
# primes1 = []
#
# for i in range(n1,n2+1):
#     if(isPrime(i)):
#         primes1.append(i)
#
# if(len(primes1) == 0):
#     ans = 0
#
# else:
#     primes2= []
#     for i in range(len(primes1)):
#         for j in range(len(primes1)):
#             if(i!=j):
#                 combination = int(str(primes1[i]) + str(primes1[j]))
#                 if(isPrime(combination)):
#                     primes2.append(combination)
#
#     primes2 = list(set(primes2))
#
#     if(len(primes2) == 0):
#         ans = 0
#     else:
#         x = min(primes2)
#         y = max(primes2)
#         ans = fibonacci(x,y,len(primes2))
#
# print(ans)

arr = list(map(int, input().split()))
print(arr)
